### PR TITLE
Getpost methods and dummy data

### PR DESCRIPTION
**Pull request by Sharukap**

### CODE REVIEW

File: backend/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+public
```
Comment (yashodgayashan) [inline backend/.gitignore:2]:
```suggestion
public

```

File: Dummy data/dm data admins.txt [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (yashodgayashan) [inline Dummy data/dm data admins.txt:1]: Please check the nosql file extension and add it to there

File: backend/app.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+var createError = require('http-errors');
+var express = require('express');
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+
+
+var studentRouter = require('./routes/studentRouter');
+var adminRouter = require('./routes/adminRouter');
+const mongoose=require('mongoose');
```
Comment (yashodgayashan) [inline backend/app.js:10]:
```suggestion
const mongoose = require('mongoose');
```

File: backend/app.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+var createError = require('http-errors');
+var express = require('express');
+var path = require('path');
+var cookieParser = require('cookie-parser');
+var logger = require('morgan');
+
```
Comment (yashodgayashan) [inline backend/app.js:6]: Please remove unnecessary spacing.

File: backend/app.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+var studentRouter = require('./routes/studentRouter');
+var adminRouter = require('./routes/adminRouter');
+const mongoose=require('mongoose');
+const Students=require('./models/students');
```
Comment (yashodgayashan) [inline backend/app.js:11]:
```suggestion
const Students = require('./models/students');
```

File: backend/app.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+const url='mongodb://localhost:27017/conFusion';
```
Comment (yashodgayashan) [inline backend/app.js:12]:
```suggestion
const url = 'mongodb://localhost:27017/conFusion';
```

File: backend/app.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+const connect=mongoose.connect(url);
```
Comment (yashodgayashan) [inline backend/app.js:13]:
```suggestion
const connect = mongoose.connect(url);
```

File: backend/app.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+connect.then((db)=>{
```
Comment (yashodgayashan) [inline backend/app.js:14]:
```suggestion
connect.then((db) => {
```

File: backend/app.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+	console.log('connected to server');
+},(err)=>{console.log(err);});
```
Comment (yashodgayashan) [inline backend/app.js:16]:
```suggestion
},(err) => {console.log(err);});
```

File: backend/app.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+var app = express();
+
+// view engine setup
```
Comment (yashodgayashan) [inline backend/app.js:20]:
```suggestion
// View engine setup.
```

File: backend/app.js [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+app.set('views', path.join(__dirname, 'views'));
+app.set('view engine', 'jade');
+
+app.use(logger('dev'));
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+//app.use(cookieParser());
```
Comment (yashodgayashan) [inline backend/app.js:27]: Remove unnecessary comments.

File: backend/app.js [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+//app.use(express.static(path.join(__dirname, 'public')));
+
+
+app.use('/students', studentRouter);
+
+app.use('/students', adminRouter);
+// catch 404 and forward to error handler
+app.use(function(req, res, next) {
```
Comment (yashodgayashan) [inline backend/app.js:35]: Use arrow functions.

File: backend/app.js [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+  next(createError(404));
+});
+
+// error handler
+app.use(function(err, req, res, next) {
+  // set locals, only providing error in development
+  res.locals.message = err.message;
+  res.locals.error = req.app.get('env') === 'development' ? err : {};
+
+  // render the error page
```
Comment (yashodgayashan) [inline backend/app.js:45]: Please remove wrong comments.

File: backend/models/students.js [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+    },  
+    password: {
+        type: String,
+        required: true,
+		unique:true,
+		minlength:12,
+		maxlength:15
+    },
+	//password should be 12-15 characters
+	email: {
```
Comment (yashodgayashan) [inline backend/models/students.js:24]: Please do allignments.

File: backend/models/students.js [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+		default:0
+	}
+	//account type:"true"(1) is for admin and "false"(0) is for student. unless adding an admin record it is not necessary to enter account type as default value will be set to false which is for student.
+},{
+    timestamps: true
+});
+
+var Students = mongoose.model('Student', studentSchema);
+
+module.exports = Students;
```
Comment (yashodgayashan) [inline backend/models/students.js:45]:
```suggestion
module.exports = Students;

```

File: backend/routes/adminRouter.js [hunk 16]
```diff
@@ -0,0 +1,47 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+
+const Students = require('../models/students');
+
+const dishRouter = express.Router();
```
Comment (yashodgayashan) [inline backend/routes/adminRouter.js:7]: What is the meaning of dishRouter.

File: backend/routes/adminRouter.js [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+        res.json(student);
+    }, (err) => next(err))
+    .catch((err) => next(err));
+})*/
+.post((req, res, next) => {
+    res.statusCode = 403;
+    res.end('Cannot create a student record with'+ req.params.studentId+'Do not include an ID when creating a new record');
+});
+
+module.exports = dishRouter;
```
Comment (yashodgayashan) [inline backend/routes/adminRouter.js:47]:
```suggestion
module.exports = dishRouter;

```

File: backend/routes/studentRouter.js [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+        res.json(student);
+    }, (err) => next(err))
+    .catch((err) => next(err));
+})
+/*.post((req, res, next) => {
+    res.statusCode = 403;
+    res.end('Cannot create a student record with'+ req.params.studentId+'Do not include an ID when creating a new record');
+})*/;
+
+module.exports = studentRouter;
```
Comment (yashodgayashan) [inline backend/routes/studentRouter.js:47]: Please add a new line at the end of the file.

```suggestion
module.exports = studentRouter;

```

File: backend/routes/studentRouterAll.js [hunk 19]
```diff
@@ -0,0 +1,80 @@
+const express = require('express');
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+
+const Students = require('../models/students');
+
+const dishRouter = express.Router();
```
Comment (yashodgayashan) [inline backend/routes/studentRouterAll.js:7]: Please do the naming properly.

File: backend/routes/studentRouterAll.js [hunk 20]
```diff
@@ -0,0 +1,80 @@
...
+    Students.findByIdAndRemove(req.params.studentId)
+    .then((resp) => {
+        res.statusCode = 200;
+        res.setHeader('Content-Type', 'application/json');
+        res.json(resp);
+    }, (err) => next(err))
+    .catch((err) => next(err));
+});
+
+module.exports = dishRouter;
```
Comment (yashodgayashan) [inline backend/routes/studentRouterAll.js:80]:
```suggestion
module.exports = dishRouter;

```

File: backend/views/error.jade [hunk 21]
```diff
@@ -0,0 +1,6 @@
+extends layout
```
Comment (yashodgayashan) [inline backend/views/error.jade:1]: Do we really need views?

File: backend/routes/adminRouter.js (hunk 16 above)
Comment (AselGunaratne) [reply backend/routes/adminRouter.js:7]: It should be adminRouter, nor dishRouter

File: backend/routes/adminRouter.js (hunk 17 above)
Comment (AselGunaratne) [reply backend/routes/adminRouter.js:47]:
```suggestion
+module.exports = adminRouter;
```

File: backend/routes/adminRouter.js [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+        console.log('Student record created', student);
+        res.statusCode = 200;
+        res.setHeader('Content-Type', 'application/json');
+        res.json(student);
+    }, (err) => next(err))
+    .catch((err) => next(err));
+});
+
+dishRouter.route('/:studentId')
+/*.get((req,res,next) => {
```
Comment (AselGunaratne) [inline backend/routes/adminRouter.js:33]:
```suggestion
adminRouter.route('/:studentId')
/*.get((req,res,next) => {
```

File: backend/routes/adminRouter.js [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+
+const Students = require('../models/students');
+
+const dishRouter = express.Router();
+
+dishRouter.use(bodyParser.json());
+
+dishRouter.route('/')
```
Comment (AselGunaratne) [inline backend/routes/adminRouter.js:11]:
```suggestion
adminRouter.route('/')
```

**Comment by yashodgayashan** [comment]
@AselGunaratne Please do not approve until he made changes. Thank you.

File: backend/views/error.jade (hunk 21 above)
Comment (Sharukap) [reply backend/views/error.jade:1]: I'll remove them

File: backend/routes/adminRouter.js (hunk 16 above)
Comment (Sharukap) [reply backend/routes/adminRouter.js:7]: Yes it should be adminRouter

File: backend/views/error.jade (hunk 21 above)
Comment (Sharukap) [reply backend/views/error.jade]: It gave an error when I removed the views so I had to add them back

**Pull request merged**